packages/amplication-data-service-generator-runner/src/utils/Tarball.ts

```diff
@@ -0,0 +1,37 @@
+import downloadHelper from "download";
+import { packument } from "pacote";
+import { PackageInstallation } from "./DynamicPackageInstallationManager";
+
+export class Tarball {
+  constructor(
+    protected readonly plugin: PackageInstallation,
+    private readonly modulesPath: string
+  ) {}
+  async download(): Promise<void> {
+    const tarball = await this.packageTarball(this.plugin);
+    const { name } = this.plugin;
+    await downloadHelper(tarball, this.modulesPath, {
+      extract: true,
+      map(file) {
+        file.path = file.path.replace("package/", `${name}/`);
+        return file;
+      },
+    });
+    return;
+  }
+
+  private async packageTarball({
+    name,
+    version,
+  }: PackageInstallation): Promise<string> {
+    const fullPackageName = `${name}@${version}`;
```

Comment:
by the way... following our conversation with @yuval-hazaz . if the package has the same name it will run over the other one